src/Microsoft.VisualStudio.ProjectSystem.Managed.UnitTests/Mocks/IWorkspaceProjectContextFactoryFactory.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Moq;
+
+namespace Microsoft.VisualStudio.LanguageServices.ProjectSystem
+{
+    internal static class IWorkspaceProjectContextFactoryFactory
```

Comment:
Have I mentioned that I hate `FactoryFactory` yet?

Reply:
We're officially enterprise-grade now :)

---
In reply to: [212745005](https://github.com/dotnet/project-system/pull/3905#discussion_r212745005) [](ancestors = 212745005)

Reply:
What would you call this?